src/grlib.c

```diff
@@ -151,7 +150,7 @@ grlib_ahbs_add (const struct grlib_ipcore *core, int irq,
     {
       ahbscores[ahbsi].start = addr;
       ahbscores[ahbsi].end = addr + ~(mask << 20) + 1;
-      ahbscores[ahbsi].mask = ~(mask << 20);
+      ahbscores[ahbsi].mask = 0x00FFFFFF;
```

Comment:
Are you sure that mask should be hardcoded? It seems to be the functions argument, and it may be reused by different devices.

Reply:
ack